sympy/simplify/tests/test_simplify.py

```diff
@@ -820,3 +820,19 @@ def test_issue_7971():
     z = Integral(x, (x, 1, 1))
     assert z != 0
     assert simplify(z) is S.Zero
+
+
+def test_issue_17141():
+    # Check that there is no RecursionError
+    assert simplify(x**(1 / acos(I))) == x**(2/(pi - 2*I*log(1 + sqrt(2))))
+    assert simplify(acos(-I)**2*acos(I)**2) == \
+           log(1 + sqrt(2))**4 + pi**2*log(1 + sqrt(2))**2/2 + pi**4/16
+    assert simplify(2**acos(I)**2) == 2**((pi - 2*I*log(1 + sqrt(2)))**2/4)
+
+    # However, for a complex number it still happens
+    if PY3:
+        raises(RecursionError, lambda: simplify(2**acos(I+1)**2))
```

Comment:
You can detect numbers with `I` in them as:
```python
>>> from sympy.core.evalf import pure_complex as f
>>> f(I)
(0, 1)
>>> f(1+2*I)
(1, 2)
>>> f(3) is None
True
```

Reply:
I think it boils down to the question in #17155 so I am hoping to get some feedback there.